chia/consensus/blockchain.py

```diff
@@ -380,6 +380,7 @@ async def _reconsider_peak(
         # Rollback sub_epoch_summaries
         self.__height_map.rollback(fork_height)
         await self.block_store.rollback(fork_height)
+        self._peak_height = None if fork_height < 0 else uint32(fork_height)
```

Comment:
one possible issue with this is that we don't restore anything in case of a failure. but we don't restore the `__height_map` either.

Reply:
yeah that is a potential issue. In the case of reverting the TX, everything that happened needs to be rolled back

Reply:
Wouldn't this put the blockchain into an inconsistent state? the DB is not committed yet, so we still have all the coins and blocks there (not rolled back yet)

Reply:
Also shouldn't line 389 be:
`while fork_height >= 0` ?

Reply:
> Wouldn't this put the blockchain into an inconsistent state? the DB is not committed yet, so we still have all the coins and blocks there (not rolled back yet)

Yes, it would. In fact, it *is* because the call to `rollback()` is never undone right now.